src/app/user-detail-viewer/commit-list/commit-list.component.ts

```diff
@@ -56,13 +82,23 @@ export class CommitListComponent implements OnInit {
 
   createPrefixSum() {
     if (this.commitList.length === 0) {
+      this.ps = new PrefixSum([], () => new CumulativeStats());
       return;
     }
     this.firstTime = toMinTime(this.commitList[0].committedDate);
 
     const commitToIndex = (commit: GithubCommit) => this.dateToIndex(commit.committedDate);
 
     this.endIndex = commitToIndex(this.commitList[this.commitList.length - 1]);
+    const prefixArr = new Array(this.endIndex + 1);
+    for (let i = 0; i < prefixArr.length; i++) {
+      prefixArr[i] = new CumulativeStats();
+    }
+    for (const commit of this.commitList) {
+      prefixArr[commitToIndex(commit)].add(new CumulativeStats(commit));
```

Comment:
I checked your `dateToIndex` function, it uses a sparse array. Tested on cs2103, it works fine without any jitters. We can go with this. 👍

Reply:
For reference, the average size of the array will be from 40-50